libs/common/src/sql-db-connection/migrations/1707306956373-AddUtilisationReportTables.ts

```diff
@@ -0,0 +1,95 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class AddUtilisationReportTables1707306956373 implements MigrationInterface {
+  name = 'AddUtilisationReportTables1707306956373';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`
+      CREATE TABLE "AzureFileInfo" (
+        "id" int NOT NULL IDENTITY(1, 1),
+        "folder" nvarchar(255) NOT NULL,
+        "filename" nvarchar(255) NOT NULL,
+        "fullPath" nvarchar(255) NOT NULL,
+        "url" nvarchar(255) NOT NULL,
+        "mimetype" nvarchar(255) NOT NULL,
+        "utilisationReportId" int NOT NULL,
+        CONSTRAINT "PK_ea6ad35b38f9b9e248b432d1a1e" PRIMARY KEY ("id")
+      )
+    `);
+
+    await queryRunner.query(`
+      CREATE UNIQUE INDEX "REL_4370b01aed42cea044ee2b1ce6" ON "AzureFileInfo" ("utilisationReportId")
+      WHERE "utilisationReportId" IS NOT NULL
+    `);
+
+    await queryRunner.query(`
+      CREATE TABLE "UtilisationData" (
+        "id" int NOT NULL IDENTITY(1, 1),
+        "facilityId" nvarchar(255) NOT NULL,
+        "exporter" nvarchar(255) NOT NULL,
+        "baseCurrency" nvarchar(255) NOT NULL,
+        "facilityUtilisation" int NOT NULL,
+        "totalFeesAccruedForTheMonth" int NOT NULL,
+        "totalFeesAccruedForTheMonthCurrency" nvarchar(255) NOT NULL,
```

Comment:
I'm happy with all of these being not null but will need to make sure in the refactor everything has a correct default value (not quite like that on mongodb right now, think if currency is empty it doesn't default to baseCurrency.
Somewhere we'll need to have it documented all these kind of edge case/defaulting values that we'll get in the DB.
Might be worth us chatting to Rich and Andy about this when we talk about schema of saying hey it's easier for us if we default if it's not necessary

Reply:
business logic it is!